update_ip/ip_getters/base.py

```diff
@@ -1,42 +1,49 @@
 import re
-import urllib2
 import socket
+import urllib2
 
 socket.setdefaulttimeout(5)
 
+
 class GetIpFailed(Exception):
     pass
 
-ip_regex= re.compile("\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}")
+
+# Check for valid IP Addresses
+ip_regex = re.compile(r"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
```

Comment:
I'm all for correctness, and agree that the current version is not giving correct results, but I think this is a bit too much - I have no idea if this is correct at all!
I think a good compromise might be leaving (or simplifying) the current regex and adding a validation function.